src/crimson/osd/osd.cc

```diff
@@ -1084,6 +1203,21 @@ seastar::future<> OSD::_handle_osd_map(Ref<MOSDMap> m)
         superblock.mounted = boot_epoch;
         superblock.clean_thru = last;
       }
+      // record new purged_snaps
+      if (superblock.purged_snaps_last == start - 1) {
+        OSDriver osdriver{&store.get_sharded_store(),
+                          pg_shard_manager.get_meta_coll().collection(),
+                          make_purged_snaps_oid()};
+        auto cct = CephContext();
+        SnapMapper::record_purged_snaps(
```

Comment:
This needs to be run in a seastar::async context.  As written, I think it will simply crash.